### PR TITLE
[M1584] Move serializer validation so it's run first during validation

### PR DESCRIPTION
**Pull request by gridcell**
- **BIG CHANGE ALERT:** if the serializer validation fails, no further validations are run and a response is returned

### CODE REVIEW

**Comment by gridcell** [comment]
[date format 500](https://trello.com/c/r4Bw6n9P/1584-date-format-500)

File: src/api/submission/protocol_validations.py [hunk 1]
```diff
@@ -243,9 +255,12 @@ class FishBeltProtocolValidation(TransectValidation):
     SAMPLE_UNIT = "fishbelt_transect"
 
     def validate(self):
-        results = [super(FishBeltProtocolValidation, self).validate()]
+        try:
```
Comment (saanobhaai) [inline src/api/submission/protocol_validations.py:258]: Would it be better/possible to put this in the `validate` method of both `TransectValidation` and `QuadratValidation`, and then these calls in individual protocol validations wouldn't have to change?:

```
        try:
            return [super().validate()]
        except SerializeValidationError:
            return ERROR
```

File: src/api/submission/writer.py [hunk 2]
```diff
@@ -77,27 +81,26 @@ def _create_sample_event(self, sample_event_data):
...
+        }
+        se = SampleEvent.objects.filter(**query_params)
+        if se.exists():
+            sample_event = se[0]
+            notes = sample_event_data.get("notes") or ""
+            if notes.strip():
+                sample_event.notes += "\n\n{}".format(notes)
+                sample_event.save()
+            return sample_event
+        return serializer.save()
```
Comment (saanobhaai) [inline src/api/submission/writer.py:103]: ok so with this refactor we should be able to delete the whole `_create_sample_event` method, right?

**Review by saanobhaai — CHANGES_REQUESTED** [review]
Two small refactor questions

**Pull request merged**